### PR TITLE
support perf site replication

### PR DESCRIPTION
**Pull request by jiuker**
support perf site replication

### CODE REVIEW

File: perf-site-replication.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// SiteReplicationperfNodeResult - stats from each server
+type SiteReplicationperfNodeResult struct {
```
Comment (poornas) [inline perf-site-replication.go:31]:
```suggestion
type SiteReplicationPerfNodeResult struct {
```

File: perf-site-replication.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	Endpoint string `json:"endpoint"`
+	TX       uint64 `json:"tx"`
+	RX       uint64 `json:"rx"`
+	Error    string `json:"error,omitempty"`
+}
+
+// SiteReplicationperfResult - aggregate results from all servers
+type SiteReplicationperfResult struct {
```
Comment (poornas) [inline perf-site-replication.go:39]:
```suggestion
type SiteReplicationPerfResult struct {
```

File: perf-site-replication.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	"net/url"
+	"time"
+)
+
+// SiteNetPerfNodeResult  - stats from each server
+type SiteNetPerfNodeResult struct {
+	Endpoint string `json:"endpoint"`
+	TX       uint64 `json:"tx"`
+	RX       uint64 `json:"rx"`
+	Error    string `json:"error,omitempty"`
```
Comment (klauspost) [inline perf-site-replication.go:35]: Don't we need a time indication here, so we know the time frame spent transferring RX/TX?

I feel that should be added for completeness, so we don't have to rely on the query to parse the result.

Comment (klauspost) [reply perf-site-replication.go:35]: Don't we measure ping latency? Or is that out of scope?

Comment (klauspost) [reply perf-site-replication.go:35]: Also add the number of concurrent connections per peer.

File: perf-site-replication.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// SiteNetPerfNodeResult  - stats from each server
+type SiteNetPerfNodeResult struct {
+	Endpoint  string `json:"endpoint"`
+	TX        uint64 `json:"tx"`
+	TxDurMs   uint64 `json:"txDurMs"`
```
Comment (harshavardhana) [inline perf-site-replication.go:34]:
```suggestion
	TXTotalDuration   time.Duration `json:"txTotalDuration"`
```

File: perf-site-replication.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	RX        uint64 `json:"rx"`
+	RxDurMs   uint64 `json:"rxDurMs"`
```
Comment (harshavardhana) [inline perf-site-replication.go:36]:
```suggestion
	RXTotalDuration   time.Duration `json:"rxTotalDuration"`
```

File: perf-site-replication.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// SiteNetPerfNodeResult  - stats from each server
+type SiteNetPerfNodeResult struct {
+	Endpoint  string `json:"endpoint"`
+	TX        uint64 `json:"tx"`
```
Comment (harshavardhana) [inline perf-site-replication.go:33]:
```suggestion
	TX        uint64 `json:"tx"` // transfer rate in bytes
```

File: perf-site-replication.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+	TxDurMs   uint64 `json:"txDurMs"`
+	RX        uint64 `json:"rx"`
```
Comment (harshavardhana) [inline perf-site-replication.go:35]:
```suggestion
	RX        uint64 `json:"rx"` // received rate in bytes 
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**